compiler/parser/src/token.rs

```diff
@@ -25,6 +25,7 @@ pub enum Tok {
         triple_quoted: bool,
     },
     Newline,
+    NonLogicalNewline,
```

Comment:
I think `PhysicalNewLine` is better tbh, better suited when someone reads the [Lexical Analysis](https://docs.python.org/3/reference/lexical_analysis.html) section of the reference. Logical line there is basically defined as `NewLine` so we're good.

Reply:
I was also initially thinking of calling it `PhysicalNewLine`, but the concept doesn't match exactly. Technically, every logical new line is also a physical new line, by the Language Reference definition. If we were issuing a `PhysicalNewLine` before or after every `Newline`, it would match, but I chose to go with tokenize.py which doesn't do that.

Reply:
Sounds reasonable enough to me, I don't think there's much point in bike-shedding something like this.